sig-security-assessments/cluster-api/self-assessment.md

```diff
@@ -0,0 +1,1099 @@
+  * _Inbound. How do users or prospective users communicate with the team?_
+  * _Outbound. How do you communicate with your users? (e.g. flibble-announce@
+    mailing list)_
+* _Ecosystem. How does your software fit into the main project and other cloud
+  native ecosystems?_
+
+#### Development Workflow
+
+Cluster API’s development workflow is documented within the repository’s
+[CONTRIBUTING.md](https://github.com/kubernetes-sigs/cluster-api/blob/main/CONTRIBUTING.md)
+
+Key processes include:
+
+* All source code is publicly available in Github
+* All pull requests trigger jobs that perform:
+  * Linting, including gosec static code analysis
+  * Unit tests
+  * E2E tests using nested Docker as a simulated infrastructure provider
+  * Checks for breaking API changes
+  * Tests for broken URLs in generated documentation
+  * PRs cannot be merged without passing tests without an explicit /override
+    from a maintainer
+* Each PR requires approval from maintainers listed in the OWNERS file
+* For all Cluster API subprojects, contributors are required to have signed the
+  CNCF CLA, but no cryptographic signatures are required for individual commits.
+* Pushes to the main branch are forbidden by convention
+* Release process is almost fully automated via Github actions
+* Checksum, signature and SPDX files are not currently generated for any
+  artifacts
+* Container images are produced automatically by Google CloudBuild, and Cluster
+  API uses the upstream Kubernetes image promotion process to promote individual
+  OCI SHAs from staging to release.
+* Deployment manifests reference image tags rather than SHA digests
+* ACLs for ownership of the repository and approvals are configured in code
+  within the OWNERS file at the top of the repository as well as configured via
+  the github.com/kubernetes/org repository.
+* For Cluster API Provider AWS since v0.6.7 and Cluster API Provider Azure since
+  v0.4.13 maintainers do GPG sign all tags, and maintainers are expected to
+  maintain their GPG keys with public keys verifiable in Github. Cluster API
+  Core repository has tags signed by Github Actions since v1.0.3
+
+#### Identified gaps in development workflow
+
+##### [DEV-01: Outdated OWNERS file](https://github.com/kubernetes-sigs/cluster-api/issues/6151)
+
+Listed maintainers of the Cluster API project are out of date and may continue
+to have administrative access to the GitHub repository as well as perform image
+promotion and access the staging Google Cloud account. A process to regularly
+review ACLs(Access Control Lists) should be instituted.
+
+Status: implemented / planned / to be implemented
+
+#### Core Infrastructure Initiative
+
+Cluster API asserts being 96% in progress with regards to
+the [Core Infrastructure Initiative (CII) badging process](https://bestpractices.coreinfrastructure.org/en/projects/5288)
+.
+
+##### DEV-02: Vulnerability report process is poorly defined
```

Comment:
This is a huge demand for a sub-project without security experts full staffed.
It would be great if there is a template process we can refer to; pushing this a little bit further, as a community does not make sense to join forces (e.g. have a common process, share the same tooling, etc) instead of re-inventing things?

Reply:
Would be definitely nice to get to a point where an individual project (core CAPI or infra providers) only have to provide a list of contacts.

Reply:
the idea for private mailing lists for security reports was rejected some time ago, but that was a time when groups were not yet under k8s.io (community managed vs google managed). @PushkarJ i think it might be worth to open a discussion with #sig-k8s-infra about that.

- one private security mailing list per SIG makes sense to me as a start- `sig-foo-security@k8s.io`
- if a subproject really needs its own security it can be `sig-foo-subproject-security@k8s.io`.
- i don't think such a mailing list will be high volume.
- SIG leads + others can be present there and DM maintainers when a security issue arrives.

there is an issue somewhere to change the SECURITY_CONTACTS mechanism.
including github IDs there is pointless as github has no real DMs(?).
there should be a mailing list or personal emails linked either from the file or from the project README.

Reply:
IIRC, there was a CI job that validates the contents for the SC file.
but you can try adding emails today instead of github IDs as a quick fix.

Reply:
100% agree with the comments here and need for repeatable process. The relevant issues on this for k/k have not made much progress afaik (See https://github.com/kubernetes/community/pull/6484)

Maybe for now, it would make sense, to do following:
- Create such a private group for cluster-lifecycle sig as @neolit123 suggested. 
     - PR process to create a group should get sig-k8s-infra's attention if they have any concerns.
- Ensure this mailing list, is listed in the README.md of all the sub-projects owned by this SIG.
     - We ask anyone who is reading to default to `security@kubernetes.io` and 
     - State clearly only emails from SRC members will be addressed / triaged on the private mailing list. 
     - This will ensure we do not have to triage something that has not gone through SRC. 

I will try to take up the org level need for such a mailing list as part of my other responsibilities but out of scope of this assessment. 

Would that work for you all?

Reply:
I discussed this in SIG Security meeting today (04/21). SRC is going to have a similar discussion on this topic soon. I think considering the scope of the assessment, having an issue (which we already do) that tracks this work, is good enough for now. Ideally, if the proposed approach goes well it will benefit Cluster API project as well as all other K8s sub-projects and we would feel good about influencing something that had wide impact :)

Reply:
Sounds good to me!